hack/images/dcos/Makefile

```diff
@@ -2,7 +2,7 @@ FROM	?= /target
 TARGET_OBJ = km kubectl
 IMAGE	?= kubernetes
 REPO	?= mesosphere/$(IMAGE)
-TAG	?= k8s-0.14.2-k8sm-0.5-dcos-dev
+DCOS_VERSION ?= dev
 SUDO	?= $(shell test "$$(whoami)" = "root" || echo sudo)
```

Comment:
Good feedback. Some thoughts:

- 'sudo make' scares me in general. I sure hope `rm -rf /` (or something
else destructive) never accidentally executes in this Makefile because I
hosed some variable.
- 'sudo docker ..do something..' fails and should abort the make process if
the user doesn't have sudo privs, or fat-fingers the password -- and
hopefully provides a useful error message.
- I like the idea of a privs-test target. /etc/sudoers config can be pretty
specific, so we'd probably have to test some docker noop like `sudo docker
ps >/dev/null` if we wanted (a slightly more reliable way) to validate
sudo-docker privs.
- not sure what the long term plan is for this Makefile. eventually we'll
probably deploy k8sm using marathon groups, but I haven't thought very far
ahead w/ respect to the evolution of this file -- working as I go. It'll
certainly want some cleanup if we decide to keep it around for the long
haul.


On Thu, May 14, 2015 at 7:38 PM, Connor Doyle <notifications@github.com>
wrote:

> In hack/images/dcos/Makefile
> <https://github.com/mesosphere/kubernetes-mesos/pull/278#discussion_r30374353>
> :
>
> > @@ -2,7 +2,7 @@ FROM	?= /target
> >  TARGET_OBJ = km kubectl
> >  IMAGE	?= kubernetes
> >  REPO	?= mesosphere/$(IMAGE)
> > -TAG	?= k8s-0.14.2-k8sm-0.5-dcos-dev
> > +DCOS_VERSION ?= dev
> >  SUDO	?= $(shell test "$$(whoami)" = "root" || echo sudo)
>
> Instead of calling sudo maybe we could make a target that can check whoami
> and fail with a descriptive message. We'd depend on that target from the
> targets that need root. I'd rather have to explicitly type sudo make ...,
> but maybe that's just me.
>
> Probably doesn't matter much if the plan is to eventually scrap this
> Makefile.
>
> Guess it doesn't matter too much if this is
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesosphere/kubernetes-mesos/pull/278/files#r30374353>.
>